node/pkg/vaa/structs.go

```diff
@@ -180,6 +180,49 @@ func ChainIDFromString(s string) (ChainID, error) {
 	}
 }
 
+func ChainIDFromInt(num uint16) (ChainID, error) {
+	switch num {
+	case 1:
+		return ChainIDSolana, nil
+	case 2:
+		return ChainIDEthereum, nil
+	case 3:
+		return ChainIDTerra, nil
+	case 4:
+		return ChainIDBSC, nil
+	case 5:
+		return ChainIDPolygon, nil
+	case 6:
+		return ChainIDAvalanche, nil
+	case 7:
+		return ChainIDOasis, nil
+	case 8:
+		return ChainIDAlgorand, nil
+	case 9:
+		return ChainIDAurora, nil
+	case 10:
+		return ChainIDFantom, nil
+	case 11:
+		return ChainIDKarura, nil
+	case 12:
+		return ChainIDAcala, nil
+	case 13:
+		return ChainIDKlaytn, nil
+	case 14:
+		return ChainIDCelo, nil
+	case 16:
+		return ChainIDMoonbeam, nil
+	case 17:
+		return ChainIDNeon, nil
+	case 18:
+		return ChainIDTerra2, nil
+	case 10001:
+		return ChainIDEthereumRopsten, nil
+	default:
+		return ChainIDUnset, fmt.Errorf("unknown chain ID: %v", num)
+	}
+}
+
```

Comment:
```suggestion
func IsValidChainId(id ChainID) bool {
	validChains := []ChainID{
		ChainIDSolana,
		ChainIDEthereum,
		ChainIDTerra,
		ChainIDBSC,
		ChainIDPolygon,
		ChainIDAvalanche,
		ChainIDOasis,
		ChainIDAlgorand,
		ChainIDAurora,
		ChainIDFantom,
		ChainIDKarura,
		ChainIDAcala,
		ChainIDKlaytn,
		ChainIDCelo,
		ChainIDMoonbeam,
		ChainIDNeon,
		ChainIDTerra2,
	}
	for chainId := range validChains {
		if id == ChainID(chainId) {
			return true
		}
	}
	return false
}
```

Much simpler, just as obvious, easier to maintain. The caveat to both of these approaches is that a developer might miss adding a new chain to that function. The solution is to simply add a call to this function somewhere important early in the watcher initialization or somewhere so it crashes early and loudly. It *guarantees* this function will be kept up to date.

Reply:
The nauce here is that I'm trying to also solve for an invalid int before we've parse it into a ChainID.  Examples here: https://github.com/certusone/wormhole/pull/1313/files#diff-a4248d958e0773722a811482448d2bad30ea20c8e8467f6a06d639e5619306d0R189

Reply:
I ended up refactoring all the chain handling to maps, so we can stop duplicating the canonical source.